### PR TITLE
Decouple rosout publisher init from node init.

### PR DESCRIPTION
**Pull request by fujitatomoya**
depends on https://github.com/ros2/rcl/pull/1065

address https://github.com/ros2/rclcpp/issues/2147

### CODE REVIEW

File: rclc_examples/src/example_executor_only_rcl.c
Comment (fujitatomoya) [inline rclc_examples/src/example_executor_only_rcl.c:91]: since this is an example, i think it would be okay not to call `rcl_logging_rosout_init_publisher_for_node`.

**Review by JanStaschulat — APPROVED** [review]
LGTM.

**Comment by fujitatomoya** [comment]
@JanStaschulat thanks for the review, but

> depends on https://github.com/ros2/rcl/pull/1065

we should wait until this dependent fix is merged. do you want me to revert this?

**Pull request merged**
